models/ldndc/R/met2model.LDNDC.R

```diff
@@ -0,0 +1,247 @@
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+##-------------------------------------------------------------------------------------------------#
+##' Converts a met CF file to a model specific met file. The input
+##' files are calld <in.path>/<in.prefix>.YYYY.cf
+##'
+##' @name met2model.LDNDC
+##' @title Write LDNDC met files
+##' @param in.path path on disk where CF file lives
+##' @param in.prefix prefix for each file
+##' @param outfolder location where model specific output is written.
+##' @param overwrite logical: replace output files if they already exist?
+##' @return OK if everything was successful.
+##' @export
+##' @author Henri Kajasilta
+##-------------------------------------------------------------------------------------------------#
+met2model.LDNDC <- function(in.path, in.prefix, outfolder, start_date, end_date, overwrite = FALSE, ...) {
+  
+  # Logger info
+  PEcAn.logger::logger.info("START met2model.LDNDC")
+  
+  # Years
+  start_year <- lubridate::year(start_date)
+  end_year <- lubridate::year(end_date)
+  
+  
+  # Check the nc-files
+  nc_file <- list.files(in.path)[grep("*.nc", list.files(in.path))]
+  
+  # Set-up the outfile that will be returned as results
+  out.file <- paste(in.prefix, format(as.Date(start_date), "%Y-%m-%d"),
+                    format(as.Date(end_date), "%Y-%m-%d"),
+                    "txt",
+                    sep = ".")
+  
+  out.file.full <- file.path(outfolder, out.file)
+  
+  # Results
+  results <- data.frame(file = out.file.full,
+                        host = PEcAn.remote::fqdn(),
+                        mimetype = "text/plain",
+                        formatname = "LDNDC_Climate",
+                        startdate = start_date,
+                        enddate = end_date,
+                        dbfile.name = out.file,
+                        stringsAsFactors = FALSE)
+  
+  
+  
+  if(length(nc_file) > 0){
+    print("At least one nc-file was found from the given path")
+  }
+  
+  
+  for(year in start_year:end_year){
+    
+    # Year
+    PEcAn.logger::logger.info(year)
+    
+    
+    old.file <- file.path(in.path, paste0(in.prefix, ".", year, ".nc"))
+    
+    
+    if(file.exists(old.file)){
+      
+      ## Open netCDF file
+      nc <- ncdf4::nc_open(old.file)
+      
+      # Data points are relational to this date
+      units <- nc$dim$time$units
+      
+      # Check that the simulation doesn't take place before there are data points
+      if(year == start_year & PEcAn.utils::datetime2cf(start_date, units, tz = "UTC") < 0){
+        PEcAn.logger::logger.severe("No data from the start date, consider postponing it")
+      }
+      
+      
+      # Convert the time fractions to be seconds by starting from the date in file's units
+      sec <- nc$dim$time$vals
+      sec <- udunits2::ud.convert(sec, unlist(strsplit(nc$dim$time$units, " "))[1], "seconds")
```

Comment:
```suggestion
      sec <- udunits2::ud.convert(sec, unlist(strsplit(units, " "))[1], "seconds")
```
as extracted above on L77